internal/services/monitor/monitor_diagnostic_categories_data_source.go

```diff
@@ -27,7 +27,14 @@ func dataSourceMonitorDiagnosticCategories() *pluginsdk.Resource {
 				ValidateFunc: azure.ValidateResourceID,
 			},
 
-			"logs": {
```

Comment:
again, we can't just remove this, you need to deprecate it and put the removal behind the 4.0 beta flag